dcicutils/ecr_utils.py

```diff
@@ -14,37 +19,51 @@ class ECRUtils(object):
         NOTE 2: the (already created) ECR repository must have the env_name in the URI
     """
 
+    # This tag is presently called "latest", but I'd like to call it "released". -kmp 12-Mar-2022
+    # For now refer to it indirectly through a variable.
+
+    IMAGE_RELEASED_TAG = 'latest'
+
+    # In many cases, the ECR repo named 'main' is where images live.
+    # There could be blue/green deploys or other multi-environment account situations where others are used.
+    DEFAULT_ECS_REPOSITORY = 'main'
+
     REGION = COMMON_REGION  # this default must match what ecs_utils.ECSUtils and secrets_utils.assume_identity use
 
-    # Will thinks this is no longer used. -kmp 14-Jul-2022
-    #
-    # ECR_LAYOUT = {
-    #     'latest-wsgi': 'Latest version of WSGI Application',
-    #     'latest-indexer': 'Latest version of Indexer Application',
-    #     'latest-ingester': 'Latest version of the Ingester Application',
-    #     'stable-wsgi': 'Stable version of WSGI Application',
-    #     'stable-indexer': 'Stable version of Indexer Application',
-    #     'stable-ingester': 'Stable version of the Ingester Application'
-    # }
+    # Typically only the recent images are what needs to be seen when doing things like adding or removing labels.
+    # But this number is quite arbitrary. Since the command to show a list will summarize how many were not shown,
+    # it should be easy to ask for a wider window.
+    IMAGE_LIST_DEFAULT_COUNT_LIMIT = 10
+
+    # This is extremely arbitrary. It shouldn't be a major efficiency matter since this is rarely used.
+    # A small number means the feature of managing multiple chunks is regularly tested and less likely to have
+    # weird bugs creep in that we don't notice until later.
+    IMAGE_LIST_CHUNK_SIZE = 25
 
     # defaults were formerly env_name='cgap-mastertest', local_repository='cgap-wsgi'
-    def __init__(self, *, env_name, local_repository, region=None):
+    def __init__(self, *, env_name=None, local_repository=None, region=None, ecr_client=None,
+                 ecs_repository=None):
         """ Creates an ECR client on startup """
-        self.env = env_name
-        self.local_repository = local_repository
-        self.client = boto3.client('ecr', region_name=region or self.REGION)
+        self.env = env_name or os.environ.get('ENV_NAME')
+        self.local_repository = local_repository  # Not sure this is even used any more. Should we deprecate it?
+        self.ecr_client = ecr_client or boto3.client('ecr', region_name=region or self.REGION)
+        self.ecs_repository = ecs_repository or self.DEFAULT_ECS_REPOSITORY
```

Comment:
Ah, OK. Though I'm just working with pre-existing names (not sure if due to me or others), so it's trying to be a compatible change. I could _add_ an `ecr_repository` init arg as an alternative and deprecate the `ecs_repository` init arg, I guess. Probably it could just be `image_repository` for that matter, which might be clearer.

Reply:
In most cases the uses were compatible since this was new in this PR. I retained some small amount of support for ecr_repository but am generally favoring image_repository right now.